### ISSUE TITLE
[NEWS] Nowy zasób do zwrotów dla kupującego

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#2065]
Udostępniliśmy dziś nowe zasoby, dzięki którym wykonasz [zwrot płatności do kupującego](https://allegro.pl/pomoc/dla-sprzedajacych/platnosc-od-kupujacego/jak-zwrocic-wplate-kupujacemu-xG71g2rkRIx) oraz pobierzesz informację o zwrotach:

[POST /payments/refunds](https://developer.allegro.pl/documentation/#tag/Payments/paths/~1payments~1refunds/post) - wykonaj zwrot płatności
[GET /payments/refunds](https://developer.allegro.pl/documentation/#tag/Payments/paths/~1payments~1refunds/get) - pobierz listę zwrotów płatności

Możesz zlecić zwrot dla płatności utworzonych po 31.07.2019.
Więcej informacji znajdziesz w naszym poradniku - [jak obsługiwać zamówienia](https://developer.allegro.pl/orders/#makeRefund).

**Comment by BABYBOOM-SERWIS** [comment]
Dlaczego mimo iż w api jest dostępna możliwość zwrotu części kwoty za produkt to nie ma takiej możliwości przez stronę internetową?

**Comment by PrzemyslawLukanowski** [comment]
W przeszłości pojawiła się już sugestia, aby tę funkcjonalność wdrożyć także na stronie internetowej. Na tę chwilę możliwość zwrotu części kwoty za produkt dostępna jest tylko przez REST API.

**Comment by j4r3kb** [comment]
Jak rozumiem jest to zasób w zamian za doSendRefundForm w WebAPI?

Dokumentacja do niego https://developer.allegro.pl/documentation/#operation/initiateRefund jest dość skąpa. Poprzedni zasób pozwalał jedynie na przesłanie dealId i liczby sztuk.
Obecnie mamy jeszcze dane odnośnie delivery, overpaid, surcharges, additionalServices.
Jeżeli mamy scenariusz zwrotu produktu przez klienta to nie wystarczy podanie samego payment id, ale też delivery (jeśli przesyłka nie była darmowa) oraz additionalServices (jeżeli była usługa dodatkowa)?

**Comment by PrzemyslawLukanowski** [comment]
Nie, doSendRefundForm dotyczy zwrotu prowizji w przypadku, kiedy transakcja nie dojdzie do skutku. Odpowiednik tej metody udostępnimy w pierwszym kwartale tego roku. Zasób REST API, o którym wspominasz, jest związany ze zwrotem płatności do kupującego (czyli np. zwracasz wpłaconą kwotę, ponieważ towar jest już niedostępny lub kupujący wpłacił za niską kwotę). Więcej na ten temat znajdziesz w [naszym poradniku](https://developer.allegro.pl/orders/#09).

**Comment by j4r3kb** [comment]
Ta metoda jest oznaczona do wyłączenia w marcu 2020, jak w takim razie mamy udostępnić klientom zamiennik którego nie ma i powstanie równo/po dacie wyłączenia metody w WebAPI?

**Comment by PrzemyslawLukanowski** [comment]
W REST API zamierzamy udostępnić ten zasób na początku marcu, tym samym przedłużymy działanie metody WebAPI do końca kwietnia. Jeżeli pojawi sie taka potrzeba, będzie planować inne rozwiązanie.

**Comment by j4r3kb** [comment]
Jedyny problem jaki widzę, to metoda z WebAPI operuje na dealId, a mając metody od zamówień przepisane na REST nie ma już dealId a jest UUID. Wygląda na to że metoda z WebAPI jest bezużyteczna?

**Comment by PrzemyslawLukanowski** [comment]
@j4r3kb Aby otrzymać dealId, możesz skorzystać z [mapowania](https://developer.allegro.pl/documentation/#operation/findMapping). Szczegóły znajdziesz w [naszym newsie](https://github.com/allegro/allegro-api/issues/731).
